content/blog/2022-07-05-serving-models-with-mlem.md

```diff
@@ -0,0 +1,283 @@
+are used to get new predicted values and their probabilities for evaluation.
+This, along with everything else in the script, is how we get the metrics for
+each experiment.
+
+</admon>
+
+After you run an experiment, there will be two new files in your repo: `clf` and
+`clf.mlem`. Make sure you add the `clf.mlem` file to your Git history with the
+following command:
+
+```dvc
+$ git add clf.mlem
+```
+
+This is so that the metadata is in your repo and ready to use with other MLEM
+commands. Now we can finally take the model file and ship it to production.
+
+## Serve the model to production
+
+There are a couple of ways you can do this with MLEM:
+
+- Serve the model with [FastAPI](https://fastapi.tiangolo.com/)
+- Create a Python package
+
+_Note:_ There is an experimental option to
+[deploy the model directly to Heroku](https://mlem.ai/doc/get-started/deploying)
+although this functionality is experimental and may have breaking changes.
+
+### Serve with FastAPI
+
+If you don't have an API to work with and don't need a Python package, you can
+serve your model quickly using FastAPI with this command.
+
+```dvc
+$ mlem serve clf fastapi
+```
+
+This will run a local server and spin up an API for you so you can quickly test
+out your model without needing a development team to work on the API initially.
+
+You'll see an output like this in your terminal:
+
+```dvc
+(.venv) milecia@Milecias-MacBook-Pro-2 stale-model-example % mlem serve clf fastapi
+⏳️ Loading model from clf.mlem
+Starting fastapi server...
+💅 Adding route for /predict
+💅 Adding route for /predict_proba
+💅 Adding route for /sklearn_predict
+💅 Adding route for /sklearn_predict_proba
+Checkout openapi docs at <http://0.0.0.0:8080/docs>
+INFO:     Started server process [31916]
+INFO:     Waiting for application startup.
+INFO:     Application startup complete.
+INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
+```
+
+Then, when you go to the local URL, you'll see the documentation for how to use
+the model you created.
+
+![FastAPI ML model deployment](/uploads/images/2022-07-05/fastapi_docs.png)
+
+That's it! Now you know how to train a model, save it, and deploy to some
+external service quickly using MLEM!
+
+### Custom Python package
+
+Let's take a look at making a Python package and importing it into a
+[Flask](https://flask.palletsprojects.com/en/2.1.x/) app. To make the Python
+package, we'll run the following MLEM command.
+
+```dvc
+$ mlem build clf pip -c target=build/ -c package_name=bike_predictor
+```
+
+This takes our `clf.mlem` file and generates a Python package called
+`bike_predictor` in the `build` directory. When you look in your project, you
+should see that new `build` folder that has all of the files you need for an
+independent Python package.
+
+To build the package, you'll need to run the following command in the `build`
+directory.
+
+```dvc
+$ python -m build --wheel
```

Comment:
I think most of the target audience will know what wheels are and if not, it's ok if they have to look it up since it's not directly related to Iterative stuff. Plus we don't want to redirect them from our site too much.